crates/bgp-pkt/src/wire/deserializer/nlri/bgp_ls.rs

```diff
@@ -0,0 +1,536 @@
+        add_path: bool,
+    ) -> IResult<Span<'a>, Self, LocatedBgpLsNlriParsingError<'a>>
+    where
+        Self: Sized,
+    {
+        let (buf, path_id) = if add_path {
+            let (tmp, add_path) = be_u32(buf)?;
+            (tmp, Some(add_path))
+        } else {
+            (buf, None)
+        };
+        let (buf, nlri_type) = nom::combinator::map_res(be_u16, BgpLsNlriType::try_from)(buf)?;
+        let (buf, nlri_len) = be_u16(buf)?;
+        let (buf, data) = nom::bytes::complete::take(nlri_len)(buf)?;
+
+        let (_, value) = BgpLsNlriValue::from_wire(data, nlri_type)?;
+
+        Ok((buf, BgpLsNlri { path_id, value }))
+    }
+}
+
+impl<'a> ReadablePduWithOneInput<'a, bool, LocatedBgpLsNlriParsingError<'a>> for BgpLsVpnNlri {
+    fn from_wire(
+        buf: Span<'a>,
+        add_path: bool,
+    ) -> IResult<Span<'a>, Self, LocatedBgpLsNlriParsingError<'a>>
+    where
+        Self: Sized,
+    {
+        let (buf, path_id) = if add_path {
+            let (tmp, add_path) = be_u32(buf)?;
+            (tmp, Some(add_path))
+        } else {
+            (buf, None)
+        };
+
+        let (buf, nlri_type) = nom::combinator::map_res(be_u16, BgpLsNlriType::try_from)(buf)?;
+        let (buf, nlri_len) = be_u16(buf)?;
+        let (buf, data) = nom::bytes::complete::take(nlri_len)(buf)?;
+
+        let (data, rd) = parse_into_located(data)?;
+        let (_, nlri) = BgpLsNlriValue::from_wire(data, nlri_type)?;
+
+        Ok((
+            buf,
+            BgpLsVpnNlri {
+                path_id,
+                rd,
+                value: nlri,
+            },
+        ))
+    }
+}
+
+impl<'a> ReadablePduWithOneInput<'a, BgpLsNlriType, LocatedBgpLsNlriParsingError<'a>>
+    for BgpLsNlriValue
+{
+    fn from_wire(
+        buf: Span<'a>,
+        nlri_type: BgpLsNlriType,
+    ) -> IResult<Span<'a>, Self, LocatedBgpLsNlriParsingError<'a>>
+    where
+        Self: Sized,
+    {
+        let result = match nlri_type {
+            BgpLsNlriType::Node => {
+                let (span, nlri_value) = BgpLsNlriNode::from_wire(buf)?;
+                (span, BgpLsNlriValue::Node(nlri_value))
+            }
+            BgpLsNlriType::Link => {
+                let (span, nlri_value) = BgpLsNlriLink::from_wire(buf)?;
+                (span, BgpLsNlriValue::Link(nlri_value))
+            }
+            BgpLsNlriType::Ipv4TopologyPrefix => {
+                let (span, nlri_value) =
+                    BgpLsNlriIpPrefix::from_wire(buf, BgpLsNlriType::Ipv4TopologyPrefix)?;
+                (span, BgpLsNlriValue::Ipv4Prefix(nlri_value))
+            }
+            BgpLsNlriType::Ipv6TopologyPrefix => {
+                let (span, nlri_value) =
+                    BgpLsNlriIpPrefix::from_wire(buf, BgpLsNlriType::Ipv6TopologyPrefix)?;
+                (span, BgpLsNlriValue::Ipv6Prefix(nlri_value))
+            }
+            BgpLsNlriType::TePolicy => unimplemented!(),
+            BgpLsNlriType::Srv6Sid => unimplemented!(),
```

Comment:
Using `unimplemented!` causes the parser to fail ungracefully once it hits this path.  A better way to handle this is to add unknown variant in `BgpLsNlriValue` that takes `Vec<u8>`

Reply:
true, fixed in ce35174